Sources/SwiftParser/Parser.swift

```diff
@@ -629,3 +655,41 @@ extension Parser {
     )
   }
 }
+
+// MARK: Incremental Parsing
+extension Parser {
+  mutating func loadCurrentSyntaxNodeFromCache(for kind: SyntaxKind) -> Syntax? {
+    guard var parseLookup else {
+      return nil
+    }
+
+    let currentOffset = self.lexemes.getOffsetToStart(self.currentToken)
+    if let node = parseLookup.lookUp(currentOffset, kind: kind) {
```

Comment:
This won’t actually get any of the `SyntaxCursor` benefits because what you are modifying when calling `lookUp` below is the copy of `parseLookup` that you created above.

Could you check how the performance changes if you do the following? If there’s no performance win, we should consider removing `SyntaxCursor` entirely in a follow-up PR.

```suggestion
    guard parseLookup != nil else {
      return nil
    }

    let currentOffset = self.lexemes.getOffsetToStart(self.currentToken)
    if let node = self.parseLookup!.lookUp(currentOffset, kind: kind) {
```

Reply:
Oh, I forgot `parseLookup` is a struct. About the performance, see my comments below.